### PR TITLE
WWST-1275: Aeotec Water Sensor 6 - ZW122-A

### PR DESCRIPTION
**Pull request by joshk-trility**
Updated generic water sensor DTH with support for the water sensor.
Added missing security encapsulation functions.

### CODE REVIEW

**Comment by joshk-trility** [comment]
@greens This PR supersedes the previous Aeotec water sensor one that @aclly posted. It should also address concerns you had from that PR.
Ready for review.

File: devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy [hunk 1]
```diff
@@ -69,6 +84,16 @@ def updated() {
 	initialize()
 }
 
+def configure() {
+	if (zwaveInfo.mfr.equals("0086") && zwaveInfo.model.equals("007A")) {
+		def commands = []
+		// Tell sensor to send us battery information instead of USB power information
+		commands << encap(zwave.configurationV1.configurationSet(parameterNumber: 0x5E, scaledConfigurationValue: 1, size: 1))
+		commands << encap(zwave.wakeUpV1.wakeUpIntervalSet(seconds: 4 * 3600, nodeid: zwaveHubNodeId))
```
Comment (greens) [inline devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy]: I think 4 hours is the default we automatically configure wakeup devices to use.

File: devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy [hunk 2]
```diff
@@ -53,8 +56,20 @@ metadata {
 }
 
 def initialize() {
-	// Dome Leak Sensor sends WakeUpNotification every 12 hours. Please add zwaveinfo.mfr check when adding other sensors with different interval.
-	sendEvent(name: "checkInterval", value: (2 * 12 + 2) * 60 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	if (zwaveInfo.mfr.equals("0086"))
+		// 4 hour ping for Aeotec
+		sendEvent(name: "checkInterval", value: 4 * 60 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
```
Comment (greens) [inline devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy]: typically the formula for the checkinterval is `(wakeupinterval)*2 + 2 minutes` which lets us miss one check-in and gives us some wiggle room before marking a device offline.

File: devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy [hunk 3]
```diff
@@ -53,8 +56,20 @@ metadata {
...
+
+def installed() {
+	initialize()
+	//water alarm
+	def cmds = [ encap(zwave.notificationV3.notificationGet(notificationType: 0x05)),
+				 encap(zwave.batteryV1.batteryGet())]
+	response(cmds)
 }
 
 def installed() {
```
Comment (greens) [inline devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy]: looks like you have two `installed()` methods now.

File: devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy (hunk 1 above)
Comment (greens) [reply devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy]: The old code that did that was probably just stale.

File: devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy (hunk 2 above)
Comment (joshk-trility) [reply devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy]: Done.

File: devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy (hunk 1 above)
Comment (joshk-trility) [reply devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy]: Done.

File: devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy (hunk 3 above)
Comment (joshk-trility) [reply devicetypes/smartthings/zwave-water-sensor.src/zwave-water-sensor.groovy]: Done.

**Review by greens — APPROVED** [review]

**Pull request merged**